v2/contribute/code-style/react.mdx

```diff
@@ -6,11 +6,46 @@ hide_title: true
+    - If it depends on certain variables inside an object, do not make that object a dependency in the array. Instead, extract the variable from the object before the use effect, and put that variable in the dependency array.
     - If you want it to run it just once (similar to `componentDidMount`), provide it an empty dependency array.
-    - If applicable, make sure to return a function that runs in case the dependency array has changed (or component is unmounting). In this function, you can tell the `useEffect` logic to not change state for example (in case there is an `async / await` operation happening in it.)
+    - If applicable, make sure to return a function that runs in case the dependency array has changed (or the component is unmounting). In this function, you can tell the `useEffect` logic to not change state for example (in case there is an `async / await` operation happening in it.)
 
-- Refreaining returning a component from a function. Instead, make that function itself a functional component. This will prevent that component from unmounting and mounting during rerenders. If you must return a component from a function, then the consumer of that function should use `useRef`.
\ No newline at end of file
+- Refrain from returning a component from a function. Instead, make that function itself a functional component. This will prevent that component from unmounting and mounting during rerenders. If you must return a component from a function, then the consumer of that function should use `useRef` or `useMemo`.
+
+## Feature and Theme components
+
+Complex functionality should be implemented in feature components and theme components should only have local, simple, and strictly UI-related states. Some examples:
+
+### Feature component state:
+- Information that is created in one step of the process and used in another. I.e., `loginAttemptInfo` in passwordless sign in/up
+- General error displayed in the sign-in screen but set in the form submit method
+
+### Theme component state:
+- Current value of each form field
+- Time left until another retry is allowed
+
+### `useReducer`
+
+You should use a reducer to manage the state of a feature component if:
+- The state is complex (i.e.: more than a single error and/or status field)
+- The feature component functionality will be extracted into a custom hook and reused elsewhere.
+- If other similar feature components need to use a reducer. It's better to be consistent. (e.g.: sign in/up feature components of different auth recipes)
+
+If you used a reducer to manage the state you should:
+- Move both state and action types into `types.ts` of the recipe.
+- The "root" theme component should receive them as `featureState` and `dispatch`. Further down the component tree, they are usually event handlers and individual state variables.
+- Check out other components for naming conventions, keeping in mind that it can be renamed if it's imported elsewhere
```

Comment:
I don't understand "keeping in mind that it can be renamed if it's imported elsewhere"

Reply:
That means you don't need to name `useChildProps` as `useEmailPasswordSignInUpChildProps`, since inside the module, the naming is clear, and outside we'd rename them.

Reply:
okay